frontend/README.md

```diff
@@ -0,0 +1,389 @@
+        loop on progress
+            Image Gateway->>+Image Provider: Update Progress
+            Image Provider-->>+Main Page: Updates in progress uploads
+            Main Page-->>-User: Updates upload progress
+        end
+        Image Gateway-->-Image Provider: Finish Upload
+        par Image Provider to Image Provider
+            Image Provider->>+Image Provider: Updates Upload Map
+            Image Provider->>+Main Page: Shows Notification
+            Image Provider-->>+Main Page: Updates in progress uploads
+            Main Page-->>-User: Removes in progress upload
+        and Image Provider to Image Gateway
+            Image Provider->>+Image Gateway: Refresh List
+            Image Gateway->>+Models: Generate image models
+            Models-->>-Image Gateway: Image Models list
+            Image Gateway-->>-Image Provider: Return Images
+            Image Provider-->>+Main Page: Updates Image List
+            Main Page-->>-User: Renders Image list
+        end
+    end
+```
+
+### Edge Cases
+
+- What if the user selects an invalid asset type, e.g a video instead of an image? When displaying the native controls, the input should filter the assets to just accept the valid mimetypes, e.g image/jpg, image/png, etc. Aside from that, the file object should be validated before fetching the signed url.
+- What happens in case the upload fails? App should display the upload status as failed and allow users to retry.
+- What happens if the user moves to another side half-way during the upload? Navigation should be blocked that way if the user wants to move away there should be an alert to confirm the action, in case the user moves away the upload is lost.
+- What happens if the name of the image is too long? The name should be part of the metadata but not the actual upload name, that way we can be sure the S3 restrictions are not broken.
+- What if the asset is to big? Frontend should read the size of the file and validate if its something that can be uploaded, the backend should also specify the max size upload when signing the url, in case the asset is too big the upload will throw an error that should be displayed by the frontend.
+
+## Unit and End to End tests
+
+## Tooling
+
+### Unit testing
+
+As mentioned in the [official React docs](https://reactjs.org/docs/testing.html), they recommend to using Jest alongside React testing tools, these are by far the most suitable frameworks for testing react logic and components.
+
+I believe tools like Enzyme or Jasmine where good at some point as workarounds when it comes to DOM manipulation and such, I think the simplest way today is to go with Jest plus React Testing Tools.
+
+### End to End tests
+
+For End to end Tests the most solid framework out there has to be Cypress, it natively supports Typescript so it goes along with the Frontend structure, as well as supporting multiple browser types like Chrome, Electron, Firefox, webKit, among others.
```

Comment:
```suggestion
For End to end Tests the most solid framework currently is Cypress. It natively supports Typescript so it goes along with the Frontend structure, as well as supporting multiple browser types like Chrome, Electron, Firefox, WebKit, among others.
```
"End to End tests" in the header and "End to end Tests" here, your capitalization is inconsistent.